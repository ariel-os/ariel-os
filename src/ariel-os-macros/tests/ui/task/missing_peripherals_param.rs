#![no_main]

// FAIL: the `peripherals` parameter is required in this case
#[ariel_os::task(autostart)]
async fn main(_peripherals: Peripherals) {}

struct Peripherals;
