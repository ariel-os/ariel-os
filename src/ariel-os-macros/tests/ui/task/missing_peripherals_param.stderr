error: custom attribute panicked
 --> tests/ui/task/missing_peripherals_param.rs:4:1
  |
4 | #[ariel_os::task(autostart)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: to provide this function with peripherals, use the `peripherals` macro parameter
